addons/repair/functions/fnc_repair.sqf

```diff
@@ -177,23 +177,31 @@ if (vehicle _caller == _caller && {_callerAnim != ""}) then {
     [_caller, _callerAnim] call EFUNC(common,doAnimation);
 };
 
-//Get repair time
-_repairTime = if (isNumber (_config >> "repairingTime")) then {
-    getNumber (_config >> "repairingTime");
-} else {
-    if (isText (_config >> "repairingTime")) exitWith {
-        _repairTimeConfig = getText(_config >> "repairingTime");
-        if (isNil _repairTimeConfig) then {
-            _repairTimeConfig = compile _repairTimeConfig;
-        } else {
-            _repairTimeConfig = missionNamespace getVariable _repairTimeConfig;
-        };
-        if (_repairTimeConfig isEqualType 0) exitWith {
-            _repairTimeConfig;
+// Get repair time
+_repairTime = [
+    configFile >> "CfgVehicles" >> typeOf _target >> QGVAR(repairTimes) >> configName _config,
+    "number",
+    -1
+] call CBA_fnc_getConfigEntry;
+
+if (_repairTime < 0) then {
+    _repairTime = if (isNumber (_config >> "repairingTime")) then {
```

Comment:
Personal preference maybe but I think its better to have the if statement alone and set the _repairTime in each section.
So in the if statement you have _repairTime = getNumber() and in the else you have something else. It's hard to read what is happening like it is now.